packages/dds/tree/src/test/util/testChangeRebaser.spec.ts

```diff
@@ -0,0 +1,85 @@
+
+function commutativeRebaser<TChange>(data: {
+    compose: (changes: TChange[]) => TChange;
+    invert: (changes: TChange) => TChange;
+    rebaseAnchors: (anchor: AnchorSet, over: TChange) => void;
+}): ChangeRebaser<TChange> {
+    return {
+        rebase: (change: TChange, over: TChange) => change,
+        ...data,
+    };
+}
+
+const counterRebaser = commutativeRebaser({
+    compose: (changes: number[]) => changes.reduce((a, b) => a + b, 0),
+    invert: (change: number) => -change,
+    rebaseAnchors: (anchor: AnchorSet, over: number) => {},
+});
+
+describe("testChangeRebaser", () => {
+    it("test counter with safe integers", () => {
+        const output = verifyChangeRebaser(counterRebaser, new Set([-1, 2, 3, 0, -2, 4]), (a, b) => a === b);
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.equal(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    it("test counter with unsafe integers", () => {
+        const output = verifyChangeRebaser(counterRebaser, new Set([Number.MAX_SAFE_INTEGER, -10, 2]), (a, b) => a === b);
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.notEqual(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    it("test counter of floats with varying number of digits", () => {
+        const output = verifyChangeRebaser(
+            counterRebaser,
+            new Set([1.0, 1.22, -1.222]),
+            (a, b) => a === b,
+        );
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.notEqual(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    // This test case contains all the different "edge case" numbers
+    it("test counter with special number types", () => {
+        const output = verifyChangeRebaser(
+            counterRebaser,
+            new Set([
+                Number.NaN,
+                Number.MAX_VALUE,
+                Number.MIN_VALUE,
+                Number.POSITIVE_INFINITY,
+                Number.NEGATIVE_INFINITY,
+                Number.MIN_SAFE_INTEGER,
+                Number.MAX_SAFE_INTEGER,
+            ]),
+            (a, b) => a === b,
```

Comment:
`a === b` returns `false` then they're both `NaN` so our equivalence function needs to be updated to
```suggestion
            (a, b) => a === b || (isNaN(a) && isNaN(b)),
```
I suggest declaring it as a function somewhere in this file so that all your calls to `verifyChangeRebaser` can reuse it.